topo/node/ceos/ceos_test.go

```diff
@@ -0,0 +1,267 @@
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	ktest "k8s.io/client-go/testing"
+)
+
+type sends struct {
+	req string
+	err error
+}
+
+type expects struct {
+	resp      string
+	respSlice []string
+	err       error
+}
+
+type fakeExpect struct {
+	spawnErr error
+	expects  []expects
+	sends    []sends
+	closeErr error
+}
+
+func fakeSpawner(f *fakeExpect) func(string, time.Duration, ...expect.Option) (expect.Expecter, <-chan error, error) {
+	return func(string, time.Duration, ...expect.Option) (expect.Expecter, <-chan error, error) {
+		if f.spawnErr != nil {
+			return nil, nil, f.spawnErr
+		}
+		return f, nil, nil
+	}
+}
+
+func (f *fakeExpect) Expect(*regexp.Regexp, time.Duration) (string, []string, error) {
+	if len(f.expects) == 0 {
+		return "", nil, fmt.Errorf("out of expects")
+	}
+	resp := f.expects[0]
+	f.expects = f.expects[1:]
+	return resp.resp, resp.respSlice, resp.err
+}
+
+func (f *fakeExpect) ExpectBatch([]expect.Batcher, time.Duration) ([]expect.BatchRes, error) {
+	return nil, fmt.Errorf("Unimplemented")
+}
+
+func (f *fakeExpect) ExpectSwitchCase([]expect.Caser, time.Duration) (string, []string, int, error) {
+	return "", nil, 0, fmt.Errorf("Unimplemented")
+}
+
+func (f *fakeExpect) Send(string) error {
+	if len(f.sends) == 0 {
+		return fmt.Errorf("out of sends")
+	}
+	resp := f.sends[0]
+	f.sends = f.sends[1:]
+	return resp.err
+}
+
+func (f *fakeExpect) Close() error {
+	return f.closeErr
+}
+
+type fakeNode struct {
+	kClient    kubernetes.Interface
+	namespace  string
+	interfaces map[string]*node.Link
+}
+
+func (f *fakeNode) KubeClient() kubernetes.Interface {
+	return f.kClient
+}
+
+func (f *fakeNode) Interfaces() map[string]*node.Link {
+	return f.interfaces
+}
+
+func (f *fakeNode) Namespace() string {
+	return f.namespace
+}
+
+type fakeWatch struct {
+	e []watch.Event
+}
+
+func (f *fakeWatch) Stop() {}
+
+func (f *fakeWatch) ResultChan() <-chan watch.Event {
+	eCh := make(chan watch.Event)
+	go func() {
+		for len(f.e) != 0 {
+			e := f.e[0]
+			f.e = f.e[1:]
+			eCh <- e
+		}
+	}()
+	return eCh
+}
+
+var (
+	validPb = &topopb.Node{
+		Name: "pod1",
+		Config: &topopb.Config{
+			Cert: &topopb.CertificateCfg{
+				Config: &topopb.CertificateCfg_SelfSigned{
+					SelfSigned: &topopb.SelfSignedCertCfg{
+						CertName:   "testCert.pem",
+						KeyName:    "testCertKey.pem",
+						KeySize:    4096,
+						CommonName: "r1",
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestGenerateSelfSigned(t *testing.T) {
+	ki := fake.NewSimpleClientset(&corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod1",
+		},
+	})
+	reaction := func(action ktest.Action) (handled bool, ret watch.Interface, err error) {
+		f := &fakeWatch{
+			e: []watch.Event{{
+				Object: &corev1.Pod{
+					Status: corev1.PodStatus{
+						Phase: corev1.PodRunning,
+					},
+				},
+			}},
+		}
+		return true, f, nil
+	}
+	ki.PrependWatchReactor("*", reaction)
+	tests := []struct {
+		desc    string
+		wantErr bool
```

Comment:
Would prefer this to be a string so errdiff can check the returned error with the substring if we add more test cases to be sure that the function exists for the expected reason

Reply:
done